src/js/photoswipe.js

```diff
@@ -178,128 +179,120 @@ import ContentLoader from './slide/loader.js';
  */
 
-/** @type {PhotoSwipeOptions} */
-const defaultOptions = {
-  allowPanToNext: true,
-  spacing: 0.1,
-  loop: true,
-  pinchToClose: true,
-  closeOnVerticalDrag: true,
-  hideAnimationDuration: 333,
-  showAnimationDuration: 333,
-  zoomAnimationDuration: 333,
-  escKey: true,
-  arrowKeys: true,
-  returnFocus: true,
-  maxWidthToAnimate: 4000,
-  clickToCloseNonZoomable: true,
-  imageClickAction: 'zoom-or-close',
-  bgClickAction: 'close',
-  tapAction: 'toggle-controls',
-  doubleTapAction: 'zoom',
-  indexIndicatorSep: ' / ',
-  preloaderDelay: 2000,
-  bgOpacity: 0.8,
-
-  index: 0,
-  errorMsg: 'The image cannot be loaded',
-  preload: [1, 2],
-  easing: 'cubic-bezier(.4,0,.22,1)'
-};
-
 /**
  * PhotoSwipe Core
  */
 class PhotoSwipe extends PhotoSwipeBase {
   /**
-   * @param {PhotoSwipeOptions} options
+   * @param {PhotoSwipeOptions} [options]
    */
   constructor(options) {
     super();
 
-    this._prepareOptions(options);
+    this.options = this._prepareOptions(options || {});
 
     /**
      * offset of viewport relative to document
      *
-     * @type {{ x?: number; y?: number }}
+     * @type {Point}
      */
-    this.offset = {};
+    this.offset = { x: 0, y: 0 };
 
     /**
-     * @type {{ x?: number; y?: number }}
+     * @type {Point}
      * @private
      */
-    this._prevViewportSize = {};
+    this._prevViewportSize = { x: 0, y: 0 };
 
     /**
      * Size of scrollable PhotoSwipe viewport
      *
-     * @type {{ x?: number; y?: number }}
+     * @type {Point}
      */
-    this.viewportSize = {};
+    this.viewportSize = { x: 0, y: 0 };
 
     /**
      * background (backdrop) opacity
-     *
-     * @type {number}
      */
     this.bgOpacity = 1;
+    this.currIndex = 0;
+    this.potentialIndex = 0;
+    this.isOpen = false;
+    this.isDestroying = false;
+    this.hasMouse = false;
 
-    /** @type {HTMLDivElement} */
+    /**
+     * @private
+     * @type {SlideData}
+     */
+    this._initialItemData = {};
+    /** @type {Bounds | null} */
+    this._initialThumbBounds = null;
+
+    /** @type {HTMLDivElement | undefined} */
     this.topBar = undefined;
+    /** @type {HTMLDivElement | undefined} */
+    this.element = undefined;
+    /** @type {HTMLDivElement | undefined} */
+    this.template = undefined;
+    /** @type {HTMLDivElement | undefined} */
+    this.container = undefined;
+    /** @type {HTMLElement | undefined} */
+    this.scrollWrap = undefined;
+    /** @type {Slide | undefined} */
+    this.currSlide = undefined;
 
     this.events = new DOMEvents();
-
-    /** @type {Animations} */
     this.animations = new Animations();
-
     this.mainScroll = new MainScroll(this);
     this.gestures = new Gestures(this);
     this.opener = new Opener(this);
     this.keyboard = new Keyboard(this);
     this.contentLoader = new ContentLoader(this);
+    // initialize scroll wheel handler to block the scroll
+    this.scrollWheel = new ScrollWheel(this);
+    this.ui = new UI(this);
```

Comment:
Moving Scrollwheel and UI initialization potentially breaks stuff.

Not only it breaks third-party components that rely on UI initializing later but also breaks the current wheel pain/zoom functionality.

Possibly `ScrollWheel` class needs a check to avoid such issues.

Reply:
fixed